src/lib/openapi/mappers/feature-strategy.mapper.ts

```diff
@@ -0,0 +1,29 @@
+import { SchemaMapper } from './mapper';
+import { IFeatureStrategy } from '../../types/model';
+import { CreateStrategySchema } from '../spec/create-strategy-schema';
+import { UpdateStrategySchema } from '../spec/update-strategy-schema';
+import { FeatureStrategySchema } from '../spec/feature-strategy-schema';
+
+export class FeatureStrategyMapper
+    implements
+        SchemaMapper<
+            FeatureStrategySchema,
+            IFeatureStrategy,
+            CreateStrategySchema | UpdateStrategySchema
+        >
+{
+    fromPublic(input: FeatureStrategySchema): IFeatureStrategy {
+        return {
+            ...input,
+            id: input.id || '',
```

Comment:
I'm guessing the `|| ''`' is to make the type system happy here? I can see why this is necessary but it does make me a bit nervous. If we improve the internal types in the future would it be possible to make this go away?

Reply:
Yes, this should go away eventually. Right now the id is optional in FeatureStrategySchema so this was needed.  We could also do this ``` fromPublic(input: FeatureStrategySchema): IFeatureStrategy {
        return {
            ...input,
            id: input.id!,
            projectId: input.projectId!,
        };
    }``` 

but I think that makes it more fragile

Reply:
Ah that makes sense and sets my mind at ease. There's definitely going to be a few mismatches here and there. Happy with the approach just so long as it's possible to trim out later after we have all our types in order